script.js

```diff
@@ -0,0 +1,61 @@
+let bookCollection = {};
+let booksArray = Object.keys(bookCollection);
+
+function storeBooks() {
+  window.localStorage.setItem('books', JSON.stringify(bookCollection));
+}
+
+function createList() {
+  booksArray = Object.keys(bookCollection);
+  booksArray.forEach((book) => {
+    const removeButton = document.createElement('button');
+    const dividerLine = document.createElement('hr');
+
+    removeButton.innerHTML = 'Remove';
+    removeButton.id = book;
+
+    const bookSection = document.createElement('div');
+    const title = document.createElement('p');
+    const author = document.createElement('p');
+
+    title.innerHTML = book;
+    author.innerHTML = bookCollection[book];
+
+    bookSection.append(title, author, removeButton, dividerLine);
+    document.body.querySelector('#list-of-books').append(bookSection);
+
+    document.body.querySelector(`#${book}`).addEventListener('click', () => {
```

Comment:
If you change a data structure to an array, then this `id` can be the index in the array, so `removeBook` would use something like this: 

```js
var array = ["Item", "Item", "Delete me!", "Item"]

array.splice(2,1)
```

-> array is now `["Item", "Item", "Item"]` as the `splice` arguments remove `1` item located at an index of `2`.